### ISSUE TITLE
werf helm adds empty objects to deployment manifest

### ISSUE THREAD

**Issue by vranystepan** [werf/werf#2794]
Hi folks,
it seems that `werf helm` adds some invalid objects to release manifest. It might be caused by empty template files without any resources (caused by some conditions in teamplates).

Version: `werf v1.2.0-alpha10`

```
Error: unable to build kubernetes objects from release manifest: unable to decode "": Object 'Kind' is missing in '{"metadata":{"annotations":{"werf.io/version":"v1.2.0-alpha10"}}}
```

I was inspecting output from `helm template` and I found some noise there:

```yaml
metadata:
  annotations:
    werf.io/version: v1.2.0-alpha10
```

## steps to reproduce:

```
helm create test
cat << EOF > test/templates/empty.yaml
---
---
---
EOF
werf helm upgrade --install test ./test --wait --namespace tmp1 --create-namespace
```

**result:**

```
Error: UPGRADE FAILED: error validating "": error validating data: [apiVersion not set, kind not set]
```

**template generated:**

```
werf helm template test ./test --namespace tmp1
```

```
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    werf.io/version: v1.2.0-alpha10
  labels:
    app.kubernetes.io/instance: test
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: test
    app.kubernetes.io/version: 1.16.0
    helm.sh/chart: test-0.1.0
  name: test

---
apiVersion: v1
kind: Service
metadata:
  annotations:
    werf.io/version: v1.2.0-alpha10
  labels:
    app.kubernetes.io/instance: test
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: test
    app.kubernetes.io/version: 1.16.0
    helm.sh/chart: test-0.1.0
  name: test
spec:
  ports:
  - name: http
    port: 80
    protocol: TCP
    targetPort: http
  selector:
    app.kubernetes.io/instance: test
    app.kubernetes.io/name: test
  type: ClusterIP

---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    werf.io/version: v1.2.0-alpha10
  labels:
    app.kubernetes.io/instance: test
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: test
    app.kubernetes.io/version: 1.16.0
    helm.sh/chart: test-0.1.0
  name: test
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/instance: test
      app.kubernetes.io/name: test
  template:
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/name: test
    spec:
      containers:
      - image: nginx:1.16.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: http
        name: test
        ports:
        - containerPort: 80
          name: http
          protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: http
        resources: {}
        securityContext: {}
      securityContext: {}
      serviceAccountName: test

---
metadata:
  annotations:
    werf.io/version: v1.2.0-alpha10
---
# Source: test/templates/tests/test-connection.yaml
apiVersion: v1
kind: Pod
metadata:
  name: "test-test-connection"
  labels:
    helm.sh/chart: test-0.1.0
    app.kubernetes.io/name: test
    app.kubernetes.io/instance: test
    app.kubernetes.io/version: "1.16.0"
    app.kubernetes.io/managed-by: Helm
  annotations:
    "helm.sh/hook": test-success
spec:
  containers:
    - name: wget
      image: busybox
      command: ['wget']
      args: ['test:80']
  restartPolicy: Never
```

**Comment by vranystepan** [comment]
Perhaps I can see the root cause:

https://github.com/werf/werf/blob/32f48b5c3abc35c3d3b154d2362454df58712aa1/pkg/deploy/helm/extra_annotations_and_labels_post_renderer.go#L59

below this block, we can add something to check if the object is empty or not. I mean something like

```go
		if len(obj.UnstructuredContent()) == 0 {
			continue
		}
```

**Comment by vranystepan** [comment]
resolved in #2795

**Comment by distorhead** [comment]
Thanks! Fix release will be available in 15 min.
